internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala

```diff
@@ -125,74 +122,14 @@ final class AnalyzingCompiler(
     ()
   }
 
-  def newCachedCompiler(
+  override def newCachedCompiler(
       arguments: Array[String],
       output: Output,
       log: xLogger,
       reporter: Reporter
-  ): CachedCompiler =
-    newCachedCompiler(arguments: Seq[String], output, log, reporter)
-
-  def newCachedCompiler(
-      arguments: Seq[String],
-      output: Output,
-      log: xLogger,
-      reporter: Reporter
-  ): CachedCompiler = {
-    val (bridge, bridgeClass) = bridgeInstance(compilerBridgeClassName, log)
-    val compiler = bridge match {
-      case intf: CompilerInterface2 =>
-        intf.newCompiler(arguments.toArray[String], output, log, reporter)
-      case _ =>
-        // fall back to old reflection if CompilerInterface1 is not supported
-        invoke(bridge, bridgeClass, "newCompiler", log)(
-          classOf[Array[String]],
-          classOf[Output],
-          classOf[xLogger],
-          classOf[Reporter]
-        )(arguments.toArray[String], output, log, reporter)
-    }
-    compiler.asInstanceOf[CachedCompiler]
-  }
-
-  def doc(
-      sources: Seq[VirtualFile],
-      classpath: Seq[VirtualFile],
-      outputDirectory: Path,
-      options: Seq[String],
-      maximumErrors: Int,
-      log: ManagedLogger
-  ): Unit = {
-    val reporter = new ManagedLoggedReporter(maximumErrors, log)
-    doc(sources, classpath, outputDirectory, options, log, reporter)
-  }
-
-  def doc(
-      sources: Seq[VirtualFile],
-      classpath: Seq[VirtualFile],
-      outputDirectory: Path,
-      options: Seq[String],
-      log: Logger,
-      reporter: Reporter
-  ): Unit = {
-    val compArgs = new CompilerArguments(scalaInstance, classpathOptions)
-    val arguments =
-      compArgs.makeArguments(Nil, classpath, Some(outputDirectory), options)
-    onArgsHandler(arguments)
-    val (bridge, bridgeClass) = bridgeInstance(scaladocBridgeClassName, log)
-    bridge match {
-      case intf: ScaladocInterface2 =>
-        intf.run(sources.toArray, arguments.toArray[String], log, reporter)
-    }
-    /*
-      call("xsbt.ScaladocInterface", "run", log)(
-      classOf[Array[VirtualFile]],
-      classOf[Array[String]],
-      classOf[xLogger],
-      classOf[Reporter]
-    )(sources.toArray, arguments.toArray[String], log, reporter)
-     */
-    ()
+  ): CachedCompiler2 = {
+    val (bridge: CompilerInterface2, _) = bridgeInstance(compilerBridgeClassName, log)
```

Comment:
How do you know that the bridge instance supports `CompilerInterface2`?

Reply:
Ah.  I thought `CompilerInterface` was Zinc, not Zinc's Scala 2 bridge...

Reply:
I tried to save as much as possible here - https://github.com/sbt/zinc/pull/829/commits/10d69d54b8ecfba66d0271713e81b53e51215e03